src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -70,6 +70,14 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes()
                     response = new HttpResponseMessage(statusCode);
                     response.Content = new StringContent(body);
 
+                    // we default the Content-Type here, but we override below with any
+                    // Content-Type header the user might have set themselves
+                    if (Utility.IsJson(body))
```

Comment:
I know we need to make the content negotiation more robust here. This was a tactical fix. Probably what we need to do is use request.CreateResponse in this method, if that allows WebApi inbuilt content negotiation to take place (as you say above). We need to test all that out.